src/security_utilities_rust/src/microsoft_security_utilities_core/identifiable_secrets.rs

```diff
@@ -811,3 +814,114 @@ fn index_of(input_string: String, search_char: char) -> i32
     }
 }
 
+#[derive(Clone)]
+struct Detection {
+    start: u64,
+    end: u64,
+    length: usize,
+    redaction_token: String
+}
+
+pub struct SecretMasker {
+    pub scan: identifiable_scans::Scan
+}
+
+impl SecretMasker {
+    pub fn mask_secrets(&mut self, input: &str, default_redaction_token: Option<&str>, validate_checksum: bool) -> String {
+        if input.is_empty() {
+            return String::new();
+        }
+
+        self.scan.reset();
+
+        let input_as_bytes = input.as_bytes();
+        self.scan.parse_bytes(input_as_bytes);
+
+        let detections = self.scan.possible_matches();
+        let mut detections = detections.clone();
+
+        // Short-circuit if nothing to replace.
+        if detections.is_empty() {
+            return input.to_string();
+        }
+
+        // Merge positions into ranges of characters to replace.
+        let mut current_detections = Vec::new();
+        let mut current_detection : Option<Detection> = None;
+
+        detections.sort_unstable_by_key(|item| item.start);
+
+        for detection in detections.iter() {
+            let scan_match = detection.matches_bytes(input_as_bytes, true).unwrap();
+            let match_text = scan_match.text();
+
+            if validate_checksum {
+                let match_text_as_bytes = match_text.as_bytes();
+                let mut signature_bytes = vec![0; 3];
+                signature_bytes[0] = match_text_as_bytes[57];
+                signature_bytes[1] = match_text_as_bytes[58];
+                signature_bytes[2] = match_text_as_bytes[59];
+
+                let signature = general_purpose::STANDARD.encode(&signature_bytes);
+
+                let checksum_validation_result = try_validate_common_annotated_key(
+                    &match_text,
+                    &signature,
+                );
+
+                assert!(checksum_validation_result);
+            }
+
+            // Get c3id for the find
+            let match_text_c3id = cross_company_correlating_id::generate_cross_company_correlating_id(match_text);
+
+            let redaction_token = match default_redaction_token {
+                Some(token) => token,
+                None => &format!("SEC101/200:{}", match_text_c3id),
+            };
+
+            if current_detection.is_none() {
+                current_detection = Some(Detection {
+                    start: detection.start(),
+                    end: detection.start() + detection.len() as u64,
+                    length: detection.len(),
+                    redaction_token: redaction_token.to_string()
+                });
+                current_detections.push(current_detection.clone().unwrap());
+            }
+            else {
```

Comment:
else should be on the line above, IE: } else {

Reply:
Fixed.